client/ayon_deadline/plugins/publish/global/submit_publish_job.py

```diff
@@ -224,83 +224,43 @@ def _submit_deadline_post_job(self, instance, job, instances):
-                "SingleFrameOnly": "True",
-            },
-            # Mandatory for Deadline, may be empty
-            "AuxFiles": [],
-        }
-
-        # add assembly jobs as dependencies
+        # Collect dependent jobs
+        dependency_ids = None
         if instance.data.get("tileRendering"):
             self.log.info("Adding tile assembly jobs as dependencies...")
-            job_index = 0
-            for assembly_id in instance.data.get("assemblySubmissionJobs"):
-                payload["JobInfo"]["JobDependency{}".format(
-                    job_index)] = assembly_id  # noqa: E501
-                job_index += 1
+            dependency_ids = instance.data.get("assemblySubmissionJobs")
         elif instance.data.get("bakingSubmissionJobs"):
             self.log.info(
                 "Adding baking submission jobs as dependencies..."
             )
-            job_index = 0
-            for assembly_id in instance.data["bakingSubmissionJobs"]:
-                payload["JobInfo"]["JobDependency{}".format(
-                    job_index)] = assembly_id  # noqa: E501
-                job_index += 1
+            dependency_ids = instance.data["bakingSubmissionJobs"]
+
         elif job.get("_id"):
-            payload["JobInfo"]["JobDependency0"] = job["_id"]
-
-        for index, (key_, value_) in enumerate(environment.items()):
-            payload["JobInfo"].update(
-                {
-                    "EnvironmentKeyValue%d"
-                    % index: "{key}={value}".format(
-                        key=key_, value=value_
-                    )
-                }
-            )
-        # remove secondary pool
-        payload["JobInfo"].pop("SecondaryPool", None)
+            dependency_ids = [job["_id"]]
 
+        server_name = instance.data["deadline"]["serverName"]
         self.log.debug("Submitting Deadline publish job ...")
 
-        url = "{}/api/jobs".format(self.deadline_url)
-        auth = instance.data["deadline"]["auth"]
-        verify = instance.data["deadline"]["verify"]
-        response = requests_post(
-            url, json=payload, timeout=10, auth=auth, verify=verify)
-        if not response.ok:
-            raise Exception(response.text)
-
-        deadline_publish_job_id = response.json()["_id"]
-
-        return deadline_publish_job_id
+        deadline_addon: DeadlineAddon = (
+            context.data["ayonAddonsManager"]["deadline"]
+        )
+        return deadline_addon.submit_ayon_plugin_job(
+            server_name,
+            args,
+            job_name,
+            job["Props"]["Batch"],
+            job_type=JobType.PUBLISH,
+            department=self.deadline_department,
+            priority=priority,
+            initial_status=initial_status,
+            group=self.deadline_group,
+            pool=self.deadline_pool or None,
+            dependency_job_ids=dependency_ids,
+            output_directories=[output_dir.replace("\\", "/")],
```

Comment:
Are we lacking passing along the `output_filenames`?

Reply:
There were not passed filenames in publish job.